### PR TITLE
Gherkin Script for epic ODC-3396

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4833

Problem: Gherkin Scripts for scenarios in [ODC-3396](https://issues.redhat.com/browse/ODC-3396)

Solution: These are the gherkin scripts designed covering all the scenarios related to epic [ODC-3396](https://issues.redhat.com/browse/ODC-3396)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Feature: Topology Layout should be saved
+    As a user, working in the topology, the graphical layout should be remembered for each project
+
+
+    Background:
+        Given user is at the topolgy page
+        And user has selected namespace "aut-topology-layout-save"
+        And user has created workload "hello-openshift"
+        And user is at the topology page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:9]: Already included this statement before

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        Then user will see topology layout unchanged
+
+
+    @manual
+    Scenario: Persistance Topology Layout across page views
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected layout2
+        And user has selected Graph View
+        And user has certain zoom scale
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:50]: Value should be specific

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        And user will see zoom level unchanged
+
+    
+    @manual
+    Scenario: Persistance Topology Layout after logging out from cluster
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected layout2
+        And user has selected Graph View
+        And user has certain zoom scale
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:64]: Same as above

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        When user logged out from cluster
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:65]:
```suggestion
        When user logs out from the cluster
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        And user logged in to cluster
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:66]:
```suggestion
        And user logs in to the cluster
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        Given user has selected layout1
+        When user goes to add page
+        And user comes back to topology page
+        Then user will see topology layout unchanged
+
+    
+    @manual
+    Scenario: Topology layout2 persistance
+        Given user has selected layout2
+        When user logged out from cluster
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:39]:
```suggestion
        When user logs out from cluster
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        And user logged in to cluster
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:40]:
```suggestion
        And user logs in to the cluster
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    Scenario: Topology List View persistance
+        Given user has selected List View
+        When user logged out from cluster
+        And user logged in to cluster
+        Then user will see topology view unchanged
+
+    
+    @manual
+    Scenario: Topology layout1 persistance
+        Given user has selected layout1
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:30]: Ux defines layouts in capital i.e. Layout 1

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        When user goes to add page
+        And user comes back to topology page
+        Then user will see topology layout unchanged
+
+    
+    @manual
+    Scenario: Topology layout2 persistance
+        Given user has selected layout2
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:38]: Ux defines layouts in capital i.e. Layout 2

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        When user logged out from cluster
+        And user logged in to cluster
+        Then user will see topology layout unchanged
+
+
+    @manual
+    Scenario: Persistance Topology Layout across page views
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected layout2
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:48]: Same as above

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        Then user will see topology page unchanged
+        And user will see node location unchanged
+        And user will see zoom level unchanged
+
+    
+    @manual
+    Scenario: Persistance Topology Layout after logging out from cluster
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected layout2
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:62]: Same as above

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 12]
```diff
@@ -0,0 +1,69 @@
+Feature: Topology Layout should be saved
+    As a user, working in the topology, the graphical layout should be remembered for each project
+
+
+    Background:
+        Given user is at the topolgy page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:6]: *topology

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:9]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:50]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:64]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:65]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:39]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:40]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:30]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 9 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:38]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 10 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:48]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 12 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:6]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature (hunk 11 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:62]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        And user has zoomed in the topology to a certain amount
+        When user goes to add page
+        And user comes back to topology page
+        Then user will see topology page unchanged
+        And user will see node location unchanged
+        And user will see zoom level unchanged
+
+    
+    @manual
+    Scenario: Persistance Topology Layout after logging out from cluster
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:58]:
```suggestion
    Scenario: Persistence Topology Layout after logging out from cluster
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected Layout 2
+        And user has selected Graph View
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:62]:
```suggestion
        And user has selected Layout 2 in Graph View
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+        And user has zoomed in the topology to a certain amount
+        When user logs out from cluster
+        And user logs in to cluster
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:65]: mention the user details

**Review by makambalaji — CHANGES_REQUESTED** [review]
Format the feature file

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+        Given user has selected List View
+        When user logs out from cluster
+        And user logs in to cluster
+        Then user will see topology view unchanged
+
+    
+    @manual
+    Scenario: Topology Layout 1 persistance
+        Given user has selected Layout 1
+        When user goes to add page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:30]: *Add

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    Background:
+        Given user is at the topology page
+        And user has selected namespace "aut-topology-layout-save"
+        And user has created workload "hello-openshift"
+
+
+    @manual
+    Scenario: Topology Graph View persistance
+        Given user has selected Graph View
+        When user goes to add page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:14]: *Add

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+
+    @manual
+    Scenario: Persistance Topology Layout across page views
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected Layout 2
+        And user has selected Graph View
+        And user has zoomed in the topology to a certain amount
+        When user goes to add page
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:50]: *Add

**Comment by makambalaji** [comment]
Hi @gajanan-more  - Scripts looks good as per the standards. So I am adding approve tag for the same
 Please get the lgtm from the engineering team.
/approve

File: frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        Given user has created workload "hello-openshift"
+        And user kept workload on right top corner
+        And user has selected Layout 2 in Graph View
+        And user has zoomed in the topology to a certain amount
+        When user logs out from cluster
+        And user logs in to cluster with kubeadmin credentials
+        And user navigates to topology page
+        Then user will see topology page unchanged
+        And user will see node location unchanged
+        And user will see zoom level unchanged
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:69]: Zoom level is not persisted between sessions, only during the current session. Node locations are kept but any pan location change or zoom level will be reset.

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/topology-layout-save.feature:69]: Okay. I will update it.

**Review by jeff-phillips-18 — APPROVED** [review]

**Pull request merged**